packages/fast-components-react-msft/src/flyout/flyout.tsx

```diff
@@ -0,0 +1,219 @@
+    FlyoutProps,
+    FlyoutUnhandledProps,
+    FlyoutVerticalPosition,
+} from "./flyout.props";
+import {
+    FlyoutClassNameContract,
+    ManagedClasses,
+} from "@microsoft/fast-components-class-name-contracts-msft";
+import { DisplayNamePrefix } from "../utilities";
+
+class Flyout extends Foundation<FlyoutHandledProps, FlyoutUnhandledProps, {}> {
+    public static displayName: string = `${DisplayNamePrefix}Flyout`;
+
+    public static defaultProps: Partial<FlyoutProps> = {
+        visible: false,
+        horizontalPositioningMode: FlyoutAxisPositioningMode.uncontrolled,
+        defaultHorizontalPosition: FlyoutHorizontalPosition.uncontrolled,
+        verticalPositioningMode: FlyoutAxisPositioningMode.adjacent,
+        defaultVerticalPosition: FlyoutVerticalPosition.bottom,
+        horizontalAlwaysInView: false,
+        verticalAlwaysInView: false,
+        fixedAfterInitialPlacement: false,
+        height: "128px",
+        width: "280px",
+    };
+
+    protected handledProps: HandledProps<Required<FlyoutHandledProps>> = {
+        anchor: void 0,
+        children: void 0,
+        defaultHorizontalPosition: void 0,
+        defaultVerticalPosition: void 0,
+        describedBy: void 0,
+        visible: void 0,
+        fixedAfterInitialPlacement: void 0,
+        height: void 0,
+        horizontalAlwaysInView: void 0,
+        horizontalLockToDefault: void 0,
+        horizontalPositioningMode: void 0,
+        horizontalThreshold: void 0,
+        label: void 0,
+        labelledBy: void 0,
+        managedClasses: void 0,
+        onDismiss: void 0,
+        verticalAlwaysInView: void 0,
+        verticalLockToDefault: void 0,
+        verticalPositioningMode: void 0,
+        verticalThreshold: void 0,
+        viewport: void 0,
+        width: void 0,
+    };
+
+    private rootEl: React.RefObject<ViewportPositioner>;
+
+    constructor(props: FlyoutProps) {
+        super(props);
+
+        this.rootEl = React.createRef();
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <ViewportPositioner
+                {...this.unhandledProps()}
+                ref={this.rootEl}
+                role="dialog"
+                aria-label={this.props.label}
+                aria-labelledby={this.props.labelledBy}
+                aria-describedby={this.props.describedBy}
+                anchor={this.props.anchor}
+                defaultHorizontalPosition={
+                    ViewportPositionerHorizontalPosition[
+                        FlyoutHorizontalPosition[this.props.defaultHorizontalPosition]
+                    ]
+                }
+                defaultVerticalPosition={
+                    ViewportPositionerVerticalPosition[
+                        FlyoutVerticalPosition[this.props.defaultVerticalPosition]
+                    ]
+                }
+                disabled={!this.props.visible}
```

Comment:
Does this make sense?  When the positioner is disabled it behaves like a regular div, the positioning logic does not run but it is visible.